### PR TITLE
Add fake repository

### PR DESCRIPTION
**Pull request by Middlebrooks314**
## Summary
The `add-fake-repository` branch adds the ability to setup a stand in repository to inject into the viewcontroller for testing. 

## Future work:
- Implement the actual network call.

### CODE REVIEW

**Review by cmvandrevala — APPROVED** [review]

File: GuildedRoseLLC/Items/StaticItemRepository.swift [hunk 1]
```diff
@@ -4,20 +4,8 @@ public class StaticItemRepository : ItemRepository {
     public init() {}
     
     public func getItems(onSuccess: @escaping (_:[Item]) -> Void){
-        var items: [Item] = []
+        let items: [Item] = Item.testData
```
Comment (ericmeyer) [inline GuildedRoseLLC/Items/StaticItemRepository.swift:7]: Is `Item.testData` even needed anymore? Could we just hardcode the items in this class itself?

Having the `Item` class be responsible for the `testData` seems like a violation of SRP.

File: GuildedRoseLLC/Items/StaticItemRepository.swift [hunk 2]
```diff
@@ -4,20 +4,8 @@ public class StaticItemRepository : ItemRepository {
...
         
-        let args: String = CommandLine.arguments.count > 1 ? CommandLine.arguments[1] : ""
```
Comment (ericmeyer) [inline GuildedRoseLLC/Items/StaticItemRepository.swift:9]: Nice! It is always great to see a lot of deleted code.

File: GuildedRoseLLCSpec/ItemViewControllerSpec.swift [hunk 3]
```diff
@@ -42,13 +43,44 @@ class ItemViewControllerSpec: QuickSpec {
             
             describe("viewWillAppear") {
                 it("sets the data source to a static item list") {
+                    
+                    let itemRepository = FakeItemRepository()
+                    itemRepository.stub(items: Item.testData)
+                    controller.itemRepository = itemRepository
```
Comment (ericmeyer) [inline GuildedRoseLLCSpec/ItemViewControllerSpec.swift:48]: It looks like you are setting `controller.itemRepository` twice. Once in the setup, and once here.

What is the reason for setting it twice?

File: GuildedRoseLLCSpec/ItemViewControllerSpec.swift [hunk 4]
```diff
@@ -42,13 +43,44 @@ class ItemViewControllerSpec: QuickSpec {
...
+                it("hides the item collection view when there are no items") {
+                    
+                    let itemRepository = FakeItemRepository()
+                    itemRepository.stub(items: [])
+                    controller.itemRepository = itemRepository
+                    
+                    controller.viewDidLoad()
+                    controller.viewWillAppear(true)
+                    
+                    expect(controller.itemCollectionView.isHidden).to(beTrue())
```
Comment (ericmeyer) [inline GuildedRoseLLCSpec/ItemViewControllerSpec.swift:67]: I would split this into two tests. Once for each assertion.

File: GuildedRoseLLCSpec/ItemViewControllerSpec.swift [hunk 5]
```diff
@@ -42,13 +43,44 @@ class ItemViewControllerSpec: QuickSpec {
...
+                it("hides the item collection view when there are no items") {
+                    
+                    let itemRepository = FakeItemRepository()
+                    itemRepository.stub(items: [Item(name: "FooBar")])
+                    controller.itemRepository = itemRepository
+                    
+                    controller.viewDidLoad()
+                    controller.viewWillAppear(true)
+                    
+                    expect(controller.itemCollectionView.isHidden).to(beFalse())
```
Comment (ericmeyer) [inline GuildedRoseLLCSpec/ItemViewControllerSpec.swift:91]: I would split this into two tests. Once for each assertion.

**Review by ericmeyer — CHANGES_REQUESTED** [review]

File: GuildedRoseLLCSpec/ItemViewControllerSpec.swift (hunk 4 above)
Comment (Middlebrooks314) [reply GuildedRoseLLCSpec/ItemViewControllerSpec.swift:67]: implemented

File: GuildedRoseLLCSpec/ItemViewControllerSpec.swift (hunk 5 above)
Comment (Middlebrooks314) [reply GuildedRoseLLCSpec/ItemViewControllerSpec.swift:91]: implemented

File: GuildedRoseLLCSpec/ItemViewControllerSpec.swift (hunk 3 above)
Comment (Middlebrooks314) [reply GuildedRoseLLCSpec/ItemViewControllerSpec.swift:48]: removed line from before block... keeping tests wet for now.

File: GuildedRoseLLC/Items/StaticItemRepository.swift (hunk 2 above)
Comment (Middlebrooks314) [reply GuildedRoseLLC/Items/StaticItemRepository.swift:9]: 👍

File: GuildedRoseLLCSpec/Fakes/FakeItemRepository.swift [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import GuildedRoseLLC
+
+public class FakeItemRepository: ItemRepository {
```
Comment (ericmeyer) [inline GuildedRoseLLCSpec/Fakes/FakeItemRepository.swift:3]: What is the reason this class is made public? I would expect that the tests are the only thing using the `FakeItemRepository`.

**Review by ericmeyer — APPROVED** [review]
There are a couple small changes that I don't mind deferring to future pull requests.

**Pull request merged**